lib/vector-common/src/request_metadata.rs

```diff
@@ -145,6 +145,25 @@ impl GroupedCountByteSize {
             }
         }
     }
+
+    /// Returns `true` if we are the `Tagged` variant - keeping track of the byte sizes
+    /// grouped by their relevant tags.
+    #[must_use]
+    pub fn is_tagged(&self) -> bool {
+        match self {
+            GroupedCountByteSize::Tagged { .. } => true,
+            GroupedCountByteSize::Untagged { .. } => false,
+        }
+    }
+
+    /// Returns `true` if we are the `Untagged` variant - keeping a single count for all events.
+    #[must_use]
+    pub fn is_untagged(&self) -> bool {
+        match self {
+            GroupedCountByteSize::Tagged { .. } => false,
+            GroupedCountByteSize::Untagged { .. } => true,
+        }
```

Comment:
<!-- Thoughts represent an idea that popped up from reviewing. These comments are non-blocking by nature, but they are extremely valuable and can lead to more focused initiatives and mentoring opportunities.  -->
**💭 thought:** any reason not to change that to


```suggestion
        !is_tagged()
```

?